src/BloomExe/web/EnhancedImageServer.cs

```diff
@@ -131,6 +131,10 @@ public static SimulatedPageFile MakeSimulatedPageFileInBookFolder(HtmlDom dom, b
 			{
 				// We need to UrlEncode the single and double quote characters so they will play nicely with JavaScript. 
 				url = EscapeUrlQuotes(url);
+				// When JavaScript inserts our path into the html it replaces the three magic html characters with these substitutes.
+				// We need to modify our key so that when the JavaScript comes looking for the page its modified url will
+				// generate the right key.
+				key = key.Replace("<", "&lt;").Replace(">", "&gt;").Replace("&", "&amp;");
```

Comment:
I changed the order. I hadn't even tested the < and >, much less together with the &!